mindsdb_native/libs/phases/data_transformer/data_transformer.py

```diff
@@ -146,9 +146,7 @@ def run(self, input_data):
                 #print(self.transaction.lmd['weight_map'])
                 column_is_weighted_in_train = column in self.transaction.lmd['weight_map']
 
-                if column_is_weighted_in_train:
-                    dfs = ['input_data.validation_df']
-                else:
+                if not column_is_weighted_in_train:
                     dfs = ['input_data.train_df','input_data.test_df','input_data.validation_df']
```

Comment:
`dfs` is only defined in this `if `statement, but is referenced 6 lines below in a `for `loop independently from the `if` statement

Reply:
Oh blimey, that could have been bad... need to figure out how to modify tests for this.